src/transformers/models/dpr/modeling_tf_dpr.py

```diff
@@ -144,12 +144,12 @@ class TFDPRReaderOutput(ModelOutput):
     attentions: Optional[Tuple[tf.Tensor]] = None
 
 
-class TFDPREncoder(TFPreTrainedModel):
```

Comment:
Hmm, `TFDPREncoder` is somewhat public, so I'd prefer renaming only with Deprecation...but more importantly this would make `TFDPREncoder` looks all `TFPreTrainedModel` functionality (saving, loading weights). I don't think that is the way to go. We need to be able to compose TF model inside other models anyways at some point - there is no way around it IMO.

Reply:
Ok I'm rethinking this.